.gitlab/deploy_packages/nix.yml

```diff
@@ -146,17 +146,29 @@ deploy_packages_dogstatsd_suse_rpm-x64-7:
   variables:
     PACKAGE_ARCH: x86_64
 
-deploy_packages_dmg-x64-a7:
+.deploy_packages_dmg:
   rules:
     !reference [.on_deploy]
   stage: deploy_packages
   image: registry.ddbuild.io/ci/datadog-agent-buildimages/gitlab_agent_deploy$CI_IMAGE_GITLAB_AGENT_DEPLOY_SUFFIX:$CI_IMAGE_GITLAB_AGENT_DEPLOY
-  tags: ["arch:amd64"]
-  needs: ["agent_dmg-x64-a7"]
   before_script:
     - ls $OMNIBUS_PACKAGE_DIR
   script:
-    - $S3_CP_CMD --recursive --exclude "*" --include "datadog-agent-7*.dmg" $OMNIBUS_PACKAGE_DIR $S3_RELEASE_ARTIFACTS_URI/dmg/x86_64/
+    - $S3_CP_CMD --recursive --exclude "*" --include "datadog-agent-7*.dmg" $OMNIBUS_PACKAGE_DIR $S3_RELEASE_ARTIFACTS_URI/dmg/$PACKAGE_ARCH/
+
+deploy_packages_dmg-x64-a7:
+  extends: .deploy_packages_dmg
+  tags: ["arch:amd64"]
+  needs: ["agent_dmg-x64-a7"]
+  variables:
+    PACKAGE_ARCH: x86_64
+
+deploy_packages_dmg-arm64-a7:
+  extends: .deploy_packages_dmg
+  tags: ["arch:arm64"]
```

Comment:
I think we can use amd64 for these jobs like we do with others: https://github.com/DataDog/datadog-agent/blob/31da4bbe17af49740f04ce0023c216808aa4d84d/.gitlab/deploy_packages/deploy_common.yml#L5

Since all this does is really just uploading a package from gitlab to S3 it doesn't really need to match the architecture of the package itself.

Reply:
To clarify what I mean: this is more about moving this line to the template rather than about the specific runner to use, as it's independent of the architecture that the artifact is meant for.